### PR TITLE
two udev fixes, split out of #16859

### PR DESCRIPTION
**Pull request by poettering**
This contains two commits split out of #16859 that focus on udev's side of things and make sure we don't accidentally miss a client closing a block device and retriggering then.

### CODE REVIEW

File: src/shared/udev-util.c [hunk 1]
```diff
@@ -154,6 +154,9 @@ static int device_monitor_handler(sd_device_monitor *monitor, sd_device *device,
         assert(data->sysname || data->devlink);
         assert(!data->device);
 
+        if (device_for_action(device, DEVICE_ACTION_REMOVE))
+                return 0;
+
```
Comment (keszybz) [inline src/shared/udev-util.c:172]: Hmm, this commit needs more justification. We *may* care about initialization mostly, but we get a REMOVE event, wouldn't that mean that the waiting for the device is done too?

**Review by keszybz — COMMENTED** [review]
Looks OK, but I think we better comment on the comment is needed. I don't entirely understand the justification for the first change.

**Comment by poettering** [comment]
I force pushed a new version. Only changes: much longer comments in the code. I hope this clarifies things enough?

**Comment by keszybz** [comment]
Thanks, it is clear now.

**Pull request merged**